src/client.ts

```diff
@@ -126,8 +135,15 @@ export class Client {
       ...clientConfiguration,
       secret: this.#getSecret(clientConfiguration),
       endpoint: this.#getEndpoint(clientConfiguration),
+      logger: this.#getLogger(clientConfiguration),
     };
 
+    if (clientConfiguration && clientConfiguration.logger) {
+      this.#logger = clientConfiguration.logger;
+    } else {
+      this.#logger = new ConsoleLogHandler(LOG_LEVELS.ERROR);
+    }
```

Comment:
`getLogger` returns a logger that's turned off by default (sounds good). But then we set a logger on the client itself at error (sounds bad)? Am I reading that right?

By default we want zero logging to the console.

Reply:
oh I missed that when adding the extra log levels. It ought to be OFF.

Reply:
Made logger a required configuration and updated calls to use that instead of keeping a copy in two places.